cpp/include/cudf/utilities/resource_ref.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuda/memory_resource>
+
+namespace cudf {
+
+using host_resource_ref       = cuda::mr::resource_ref<cuda::mr::host_accessible>;
+using async_host_resource_ref = cuda::mr::async_resource_ref<cuda::mr::host_accessible>;
```

Comment:
Can you please open an RMM PR to add these aliases to RMM's `resource_ref.hpp`, and then use them instead of keeping them to libcudf?

Reply:
Done. Wasn't able to edit labels on the PR (non-breaking, cpp) though.

https://github.com/rapidsai/rmm/pull/1479